### ISSUE TITLE
[FEATURE] Historical record for completion dates of books

### ISSUE THREAD

**Issue by aaronleopold** [stumpapp/stump#207]
**Is your feature request related to a problem? Please describe.**

This feature request is a subset of some of the items discussed in https://github.com/stumpapp/stump/issues/206. Namely:

```
completed_at should be a list of dates (with at least one)... Mostly so that in the future you can look and be like "i read this on X dates" etc
```

**Describe the solution you'd like**

I don't love the idea of storing a list of dates in SQLite, since I either have to stringify it or store as bytes, so perhaps just make a small lookup table for this. Maybe rename `read_progress(es)` to `read_session(s)`, I know I've been _dying_ to rename it. The idea would then be that completion status could be tracked by something like `completion_history` or `read_history`, and `read_session` would track stuff like the page/epubcfi/etc.

**Additional context**

The changes to the schema would be somewhat small, but there would be a decent amount of changes in the API and UI required to accommodate this.

**Comment by aaronleopold** [comment]
FYI this will be landing in `experimental` in the coming weeks. I've opted to split the current `read_progresses` table into two:

1. `active_read_sessions`
2. `finished_read_sessions`

With the following stipulations:

- A user may have 1 `active_read_session` per book
- A user may have an infinite number of `finished_read_sessions` per book
- Marking a book as complete sets the `started_at` and `completed_at` timestamps to the same value
- Marking a book as unread will not clear historical record, that will have to be a separate query. Rather, it would clear the `active_read_session`.

**Comment by aaronleopold** [comment]
This is now in experimental, the migration went smoothly on my personal Stump instance 🙌 so I'll close this issue out. It will be included most likely in either 0.0.4 or 0.0.5, depending on more long-term testing.
